### ISSUE TITLE
Some identical items are separated like different

### ISSUE THREAD

**Issue by KHeket** [CleverRaven/Cataclysm-DDA#70258]
### Describe the bug

After butchering few different types of ants and web spiders - I have some chitin, and I wanted to place all excess chitin to swag bags, but if you place few spare identical chitin items in one bag it showing like "swag bag with 100 items"

<details>
  <summary>Screenshot</summary>

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/102726167/496c9277-f74b-44e0-9333-f8afd8761295)
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/102726167/30ded929-5bf0-4ef8-ad65-43bccac9132b)

</details>

But it there shouldn't be this separating, and should be look like next screenshot

<details>
  <summary>Screenshot</summary>

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/102726167/7c52db6f-61c0-4ba8-93a5-aa844e844a4f)

</details>

This situation also happens after butchering different zombies and looking at their bones, and after skinning different types of kevlar zombies

### Attach save file

[1-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13694587/1-trimmed.tar.gz)
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/102726167/d5a763c8-43ef-4dcf-9fd0-a636bf2592ed)
Character already have swag bag in hands for reproduction
1. - chitin separated similar to different items
2. Swag bag with lots of chitin from one form spiders
3. Tainted bones, separated like different items

### Steps to reproduce

1. Butcher different types of spiders(little, medium size, huge etc), zombies, skinning different types of kevlar zombies
2. See, that their chitin/bones/kevlar will be separated
3. Feel sorting pain

### Expected behavior

To not have unreasonable separation in items, what is make sorting harder

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3693 (22H2)
- Game Version: b37cb38 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by andrei8l** [comment]
They are different variants. I wrote a feature to address this (#63270) but it's only enabled in `E`at and `a`ctivate UIs due to performance concerns. And now we're kinda waiting on the imgui PR.

**Comment by KHeket** [comment]
> They are different variants

They are not different variants. Text and parameters are identical

<details>
  <summary>Spoiler</summary>

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/102726167/f0003e05-50a1-4e1a-9b59-367e40c8b7a6)

</details>

**Comment by CoroNaut** [comment]
The problem might be the inventory letter that was assigned to them when placed into the inventory:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/484914e1-5911-4023-8af0-441c5807bf4b)

There is also an extra space: "...but mutated.  The inner...".

Also additional problem that already has an issue #69601: Using item compare window, the game seems to think there is a difference in their typical damage per stamina as well, since it is colored: 
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24307650/3855d17f-ef90-42b4-80f0-dc9870ff2b06)

**Comment by andrei8l** [comment]
> > They are different variants
> 
> They are not different variants. Text and parameters are identical

You're right; sorry. CoroNaut is right, these items have different corpse IDs. Not sure why they have corpse IDs at all...

**Comment by Irets** [comment]
I can remember dissecting a large pile of zombies producing multiple piles of the same dissection produce. (tainted bone marrow, butchery refuse, etc)
Maybe this has something to do with corpse IDs too?
